client/src/components/Multiselect.tsx

```diff
@@ -380,85 +358,70 @@ export class Multiselect extends React.Component<IMultiselectProps, any> {
-      return (
-        <React.Fragment key={obj}>
-          <li className="groupHeading" style={style["groupHeading"]}>
-            {obj}
-          </li>
-          {groupedObject[obj].map((option, i) => {
-            const isSelected = this.isSelectedValue(option);
-            return (
-              <li
-                key={`option${i}`}
-                style={style["option"]}
-                className={`groupChildEle option ${isSelected ? "selected" : ""} ${
-                  this.fadeOutSelection(option) ? "disableSelection" : ""
-                } ${this.isDisablePreSelectedValues(option) ? "disableSelection" : ""}`}
-                onClick={() => this.onSelectItem(option)}
-              >
-                {showCheckbox && (
-                  <input type="checkbox" className={"checkbox"} readOnly checked={isSelected} />
-                )}
-                {this.props.optionValueDecorator(
-                  isObject ? option[displayValue] : (option || "").toString(),
-                  option
-                )}
-              </li>
-            );
-          })}
-        </React.Fragment>
-      );
-    });
-  }
-
   renderNormalOption() {
-    const { isObject = false, displayValue, showCheckbox, style } = this.props;
+    const { isObject = false, id, displayValue, showCheckbox, style } = this.props;
     const { highlightOption } = this.state;
     return this.state.options.map((option, i) => {
       const isSelected = this.isSelectedValue(option);
       return (
         <li
           key={`option${i}`}
+          id={`${id || "option"}-item-${i}`}
           style={style["option"]}
           className={`option ${isSelected ? "selected" : ""} ${
             highlightOption === i ? `highlightOption highlight` : ""
           } ${this.fadeOutSelection(option) ? "disableSelection" : ""} ${
             this.isDisablePreSelectedValues(option) ? "disableSelection" : ""
           }`}
           onClick={() => this.onSelectItem(option)}
+          onKeyDown={this.onEscKeyCloseOptionList}
+          role="option"
+          aria-selected={isSelected}
+          aria-label={isObject ? option[displayValue] : (option || "").toString()}
         >
           {showCheckbox && (
-            <input type="checkbox" readOnly className={`checkbox`} checked={isSelected} />
-          )}
-          {this.props.optionValueDecorator(
-            isObject ? option[displayValue] : (option || "").toString(),
-            option
+            <input
+              id={`option-check-${i}`}
+              type="checkbox"
+              readOnly
+              className="checkbox"
+              checked={isSelected}
+              onKeyDown={this.onEscKeyCloseOptionList}
+            />
           )}
+          <span>
+            {this.props.optionValueDecorator(
+              isObject ? option[displayValue] : (option || "").toString(),
+              option
+            )}
+          </span>
```

Comment:
```suggestion
          <label id={LABEL_ID} for={`option-check-${i}`}>
            {this.props.optionValueDecorator(
              isObject ? option[displayValue] : (option || "").toString(),
              option
            )}
          </label>
```

Reply:
i think any interactive button with labels should have an explicit label tag! 

I also thought perhaps this would fix the screenreader reading "BLANK" for each checkbox option, but changing to semantic html alone didn't fix our issue.

Reply:
ah ok, thanks! I wasn't sure it needed it because it was reading the text of the `li` and I was worried it would then read that twice. 
yeah I think the "blank" is actually it just announcing that the search box is empty every time you arrowkey navigate through the option list because that textbox is supposed to stay active (focused, targeted? not sure the word) so you can type to narrow it more at any time. 
Really am regretting picking this multiselect library initially, and not testing it out for accessibility at the start - won't do that again!

Reply:
no I think you're right, it probably should be there, since it wasn't reading it properly on browserstack/windows, so maybe that was just a quirk of OSC screen reader.